src/github.rs

```diff
@@ -1046,42 +1045,94 @@ pub struct Query<'a> {
     pub exclude_labels: Vec<&'a str>,
 }
 
+fn quote_reply(markdown: &str) -> String {
+    if markdown.is_empty() {
+        String::from("*No content*")
+    } else {
+        format!("\n\t> {}", markdown.replace("\n", "\n\t> "))
+    }
+}
+
 #[async_trait]
 impl<'q> IssuesQuery for Query<'q> {
     async fn query<'a>(
         &'a self,
         repo: &'a Repository,
+        include_fcp_details: bool,
         client: &'a GithubClient,
     ) -> anyhow::Result<Vec<crate::actions::IssueDecorator>> {
         let issues = repo
             .get_issues(&client, self)
             .await
             .with_context(|| "Unable to get issues.")?;
-
         let issues_decorator: Vec<_> = issues
-            .iter()
-            .map(|issue| crate::actions::IssueDecorator {
-                title: issue.title.clone(),
-                number: issue.number,
-                html_url: issue.html_url.clone(),
-                repo_name: repo.name().to_owned(),
-                labels: issue
-                    .labels
-                    .iter()
-                    .map(|l| l.name.as_ref())
-                    .collect::<Vec<_>>()
-                    .join(", "),
-                assignees: issue
-                    .assignees
-                    .iter()
-                    .map(|u| u.login.as_ref())
-                    .collect::<Vec<_>>()
-                    .join(", "),
-                updated_at: crate::actions::to_human(issue.updated_at),
+            .into_iter()
+            .map(|issue| async move {
+                // TODO: this seems like a *really* bad approach, but I'm not sure how to make it better
+                let fcp_map = crate::rfcbot::get_all_fcps().await?;
```

Comment:
Concern #1 — how do I get this function to only run once but still allow for it to return a `HashMap` that gets moved into an `async move`?

Reply:
can you not just move this line out of the `saync move`, like so?

```rust
let fcp_map = crate::rfcbot::get_all_fcps().await?;
let issues_decorator: Vec<_> = issues
    .into_iter()
    .map(|issue| async move {
        fcp_map[foo];
    })
    .collect();
```

Reply:
I guess not-- but what error do you get when you do so?

Reply:
also, do we want to compute this is `include_fcp_details` is false?

Reply:
When I move it out, I get the following error:

```txt
error[E0507]: cannot move out of `fcp_map`, a captured variable in an `FnMut` closure
    --> src/github.rs:1072:37
     |
1069 |            let fcp_map = crate::rfcbot::get_all_fcps().await?;
     |                ------- captured outer variable
...
1072 |                .map(|issue| async move {
     |  ___________________-__________________^
     | | __________________|
     | ||
1073 | ||                 let fcp_details = if include_fcp_details {
1074 | ||                     let repository_name = if let Some(repo) = issue.repository.get() {
1075 | ||                         repo.repository.clone()
...    ||
1086 | ||                     if let Some(fcp) = fcp_map.get(&key) {
     | ||                                        -------
     | ||                                        |
     | ||                                        move occurs because `fcp_map` has type `HashMap<std::string::String, FullFCP>`, which does not implement the `Copy` trait
     | ||                                        move occurs due to use in generator
...    ||
1131 | ||                 })
1132 | ||             })
     | ||             ^
     | ||_____________|
     | |______________captured by this `FnMut` closure
     |                move out of `fcp_map` occurs here
```

I'll try to reproduce w/ a simpler use case in playground as well.

Reply:
let fcp_map = crate::rfcbot::get_all_fcps().await?;

Reply:
And we do not want to compute this if `fcp_details` is false, that's a good catch

Reply:
Was able to work around this by changing the `issues.iter().map( ... )` call into a `for issue in issues`

See changes in c7d2261... is there anything wrong with this? Seems altogether worth is to only call the rcbot endpoint one time

Reply:
Able to replicate the error in [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a4631b395d8693a21552e7a2f472e167)